### PR TITLE
p2p: pass maxConns for MaxPeers during node setup

### LINKED ISSUE #8683: p2p: Peers don't get pruned as configured
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source): master and v0.35.x

**What happened**:
Configured max connections to 100, but peers don't get pruned until there are 1000 peers.

**What you expected to happen**:
peers to be pruned after connecting to 100

**How to reproduce it** (as minimally and precisely as possible):

1. configure a node with a smaller number of max connections than nodes in the network. 
2. Wait until peers are exchanged. 
3. Query for `net_info`

**Config (you can paste only the changes you've made)**:
```
# Maximum number of connections (inbound and outbound).
max-connections = 100
```
**node command runtime flags**: N/A

**Anything else we need to know**:
The bug originates from passing 1000 instead of the configured max connections
https://github.com/tendermint/tendermint/blob/ce6485fa709c05aed51fedf492a822f1481ed74e/node/setup.go#L230

but I'm curious about the difference between `MaxPeers` and `MaxConnected` in the peer manager options. Don't these mean the same things? afaict, `MaxPeers` is only used in pruning peers and determining peer ratio, perhaps we can just get rid of it and only use `MaxConnected`?

### PR DESCRIPTION
**Pull request by evan-forbes**
This PR changes the node setup so that it passes the configured max connections instead of what I assume is accidental constant of 1000. As discussed in #8683, ~~we might want to instead take this opportunity to remove `MaxPeers` or `MaxConnected` from the peer manager options and only use one of those values.~~

Closes #8683

### CODE REVIEW

**Comment by evan-forbes** [comment]
for more context: @tychoish and @williambanfield  found that nodes in the mamaki testnet were falling behind due to congestion on the mutexes in the consensus state, and if we could keep the peers down, then we shouldn't have an issue. Since the peers weren't actually being pruned until there were 1000, the issue was still popping up, despite their fixes. 

With this change, the nodes I'm running on the mamaki testnet seem to working fine provided their max connections are set to a reasonable value

**Review by tychoish — APPROVED** [review]

**Comment by evan-forbes** [comment]
looks like some tests are failing due to this check, but I'm usure of the difference between `MaxConnected` and `MaxPeers` so I'm sure how to change this?
https://github.com/tendermint/tendermint/blob/ce6485fa709c05aed51fedf492a822f1481ed74e/internal/p2p/peermanager.go#L169-L174

**Review by williambanfield — APPROVED** [review]

**Comment by tychoish** [comment]
my only slight concern here is that if a peer has maxconns’ number of connections and then a peer (or multiple) peers disconnect, then that peer will have to wait for pex to get more peers to store, and that’s probably not great. (assuming I’m correct about the way that maxpeers works, you probably know more about this.) maybe the right answer is to have some multiple of maxconns for this setting (2? 4?)

**Comment by cmwaters** [comment]
I tend to agree with @tychoish. You want your peer store to contain more peers than what is the maximum you can connect with so you can quickly dial new peers when connections drop. 1000 as a constant isn't probably right but I think this should be a multiple of `maxconns`. I'd be in favour of making it `maxUpgradeConns + 2 * maxConns`

**Comment by evan-forbes** [comment]
> then that peer will have to wait for pex to get more peers to store, and that’s probably not great.

ahh I see! `maxUpgradeConns + 2 * maxConns` it is https://github.com/tendermint/tendermint/pull/8684/commits/68eb230151b75b554cefaa58b6eb0d46db1c5789

File: node/setup.go [hunk 1]
```diff
@@ -223,11 +223,13 @@ func createPeerManager(
 		maxConns = 64
 	}
 
+	maxUpgradeConns := uint16(4)
```
Comment (creachadair) [inline node/setup.go:226]:
```suggestion
	const maxUpgradeConns = 4
```

**Review by creachadair — APPROVED** [review]

**Pull request merged**